PP-4617 Dockerize Node.js scripts

kbottla opened the pull request:
## WHAT

- Dockerised Node.js & build scripts so Jenkins doesn't need to be aware of Node.js / can be independent of the version

## HOW
- run `./build.sh`
- `open /build/index.html` - to view generated API docs

kbottla committed:
PP-4617 Dockerize Node.js scripts

- Dockerised Node.js & build scripts so Jenkins doesn't need to be aware of Node.js / can be independent of the version

rhowe-gds commented on .dockerignore:
```diff
@@ -0,0 +1,25 @@
+build-local.sh
```
Sort this list?

kbottla replied:
👍

rhowe-gds commented on build.sh:
```diff
@@ -37,18 +35,9 @@ else
-echo -e "${green}Built pay-api-docs static files...${NC}"
+
+# -- Build docs inside docker container --
+docker build --rm -t govukpay/pay-api-docs-generator:local .
```
This is rather racy - if we run two jobs at the same time you can't guarantee what image the following `docker run` picks up

rhowe-gds replied:
I don't think you're going to make this non-racy very easily so I'm OK with this as is

kbottla replied:
Updated to use image ID generated by docker instead. Image should now always be unique to the job

rhowe-gds commented on docker-startup.sh:
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
```
`set -eu`

kbottla replied:
👍  Updated

kbottla committed:
PP-4617 Build docker without tagging

- Build docker images without tagging (to avoid conflict between multiple jobs) and use image ID generated by Docker to build docs

kbottla committed:
PP-4617 Parameterise JenkinsFile for `PublicApiBranch`
